File: 22.Automated-Testing-Unit-Integration-and-E2E/code/typescript/anatomy_of_a_test_1.ts
// In jest or vitest
function applyDiscount(cartTotal: number, percent: number): number {
    return cartTotal - (cartTotal * percent / 100);
}

describe('applyDiscount', () => {
    it('applies the discount correctly', () => {
        // Arrange
        const total = 200;
        const percent = 10;
        // Act
        const result = applyDiscount(total, percent);
        // Assert
        expect(result).toBe(180);
    });
});

File: 22.Automated-Testing-Unit-Integration-and-E2E/code/rust/anatomy_of_a_test_1.rs
fn apply_discount(cart_total: u32, percent: u32) -> u32 {
    cart_total - (cart_total * percent / 100)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_apply_discount() {
        // Arrange
        let total = 200;
        let percent = 10;
        // Act
        let result = apply_discount(total, percent);
        // Assert
        assert_eq!(result, 180);
    }
}
